components/public-api/gitpod/v1/configuration.proto

```diff
@@ -33,6 +33,8 @@ message PrebuildSettings {
   int32 prebuild_interval = 4;
   string workspace_class = 5;
   PrebuildTriggerStrategy trigger_strategy = 6;
+  // full_clone determines if the entire repository should be cloned.
+  bool full_clone = 7;
```

Comment:
I kind of wish we had something like `ClonePreferences` wrapping this, so that we don't just pollute `PrebuildSettings`. Maybe it's not worth it, and we'll never even think about adding another prop in there, but topologically it makes sense to me, and it adds some kind of future proofing without the potential need to break this prop.

Reply:
Not necessarily; to me, both `workspace_class` is not worth abstracting any further and `branch_matching_pattern` I must say I kind of wish it was under a `PrebuildTrigger` message, which would also include `branch_strategy` and `trigger_strategy`.

Reply:
I don't understand, do you mean to use `CloneSettings` instead of `bool`? 
```
message CloneSettings {
  UNSPECIFIED
  DEPTH_ONE
  FULL_CLONE
} 
```

If so, we tested before start work on this PR that, both `depth (ws-daemon) + fetch (supervisor)` and `depth (ws-daemon) + fetch depth (supervisor)` are slower than `fully clone (ws-daemon)`

Reply:
Sorry for the confusion here, I mean that we'd essentially wrap `full_clone` in a `CloneSettings` message, to which we could potentially add more setings down the line.

Reply:
like 
```protobuf
message CloneSettings {
  full_clone bool = 1;
}
```

Reply:
yeah, that's what I mean, it's the same case with `workspace_class` that there is no other options we could set.

And if we want to re-struct this message, may make combine `clone_setting` (full_clone) and `workspace_class` together as `policies` looks more align to what we have in organization settings? because they should be in the same layer -> how we configure a prebuild workspace

```
message PrebuildPolicies {
  string workspace_class = 5;
  bool full_clone = 7;
}
message PrebuildTrigger {
  PrebuildTriggerStrategy trigger_strategy = 6;
  string branch_matching_pattern = 2;
  BranchMatchingStrategy branch_strategy = 3;
  int32 prebuild_interval = 4;
}
```

Reply:
Personally, I don't like to have lots of message layers, but use enum instead of bool would be more clear

Reply:
Uh, good idea!